docs/getting-started-guides/scratch.md

```diff
@@ -642,49 +624,57 @@ If you are using the HTTPS approach, then set:
-
-```
+Here are some apiserver flags you may need to set:
+  - `--cloud-provider=` see [cloud providers](#cloud-providers)
+  - `--cloud-config=` see [cloud providers](#cloud-providers)
+  - `--address=${MASTER_IP}`.
+   - or `--bind-address=127.0.0.1` and `--address=127.0.0.1` if you want to run a proxy on the master node.
+  - `--cluster-name=$CLUSTER_NAME`
+  - `--service-cluster-ip-range=$SERVICE_CLUSTER_IP_RANGE`
+  - `--etcd-servers=http://127.0.0.1:4001`
+  - `--tls-cert-file=/srv/kubernetes/server.cert`
+  - `--tls-private-key-file=/srv/kubernetes/server.key`
+  - `--admission-control=$RECOMMENDED_LIST`
+    - See [admission controllers](../admin/admission-controllers.md) for recommended arguments.
+  - `--allow-privileged=true`, only if you trust your cluster user to run pods as root.
 
-Then try to connect to the apiserver:
+If you are following the firewall-only security approach, then use these arguments:
+- `--token-auth-file=/dev/null`
+- `--insecure-bind-address=$MASTER_IP`
+- `--advertise-address=$MASTER_IP`
 
-```console
+If you are using the HTTPS approach, then set:
+- `--client-ca-file=/srv/kubernetes/ca.crt`
+- `--token-auth-file=/srv/kubernetes/known_tokens.csv`
+- `--basic-auth-file=/srv/kubernetes/basic_auth.csv`
+
+This pod mounts several node file system directories using the  `hostPath` volumes.  Their purposes are:
+- The `/etc/ssl` mount allows the apiserver to find the SSL root certs so it can
+  authenticate external services, such as a cloud provider.
+  This is not required if you do not use a cloud provider (e.g. bare-metal).
+- The `/srv/kubernetes` mount allows the apiserver to read certs and credentials stored on the
+  node disk.  These could instead be stored on a persistend disk, such as a GCE PD, or baked into the image.
+- Optionally, you may want to mount `/var/log` as well and redirect output there (not shown in template).
+  Do this if you prefer your logs to be accessible from the root filesystem with tools like journalctl.
 
-$ echo $(curl -s http://localhost:8080/healthz)
-ok
-$ curl -s http://localhost:8080/api
-{
-  "versions": [
-    "v1beta3",
-    "v1"
-  ]
-}
+*TODO* document proxy-ssh setup.
 
-```
+##### Cloud Providers
 
-If you have selected the `--register-node=true` option for kubelets, they will now being self-registering with the apiserver.
-You should soon be able to see all your nodes by running the `kubect get nodes` command.
-Otherwise, you will need to manually create node objects.
+if cloud providerrequires a config file (GCE, AWS). If so, need to put config file into apiserver image or mount through hostPath.
+    - options are `aws`, `gce`, `mesos`, `openshift`, `ovirt`, `rackspace`, `vagrant`, or unset.
+    - unset used for e.g. bare metal setups.
+    - support for new IaaS is added by contributing code [here](../../pkg/cloudprovider/)
+  - `--cloud-config=` set if cloud provider requires a config file.
```

Comment:
Yeah, I'm going to stop making detailed comments on this list. Please review it, as a bunch of things should be sub-bullets.